### ISSUE TITLE
[Bug] iOS safe area is working incorrectly on page which has HasNavigationBar=false

### ISSUE THREAD

**Issue by Mikilll94** [xamarin/Xamarin.Forms#9872]
### Description

When changing orientation to landscape on iPhone 8, the iOS safe area is working incorrectly on page which is inside NavigationPage and has set `HasNavigationBar=false`.

Here it is shown on the GIF

![safe_area_bug](https://user-images.githubusercontent.com/17653648/76110687-3e8d1300-5fdf-11ea-99de-876807b7358c.gif)

As you can see when I am rotating iPhone to landscape mode, at the bottom of the page there is a white bar. In portrait mode this white bar is not present.  

![image](https://user-images.githubusercontent.com/17653648/76110644-274e2580-5fdf-11ea-8778-cdb470ae88bc.png)

### Steps to Reproduce

1. Run this project on iPhone 8 (device or simulator)

2. Rotate the device/simulator to landscape orientation.
3. As you can see on the bottom of the page there is a white bar.

### Expected Behavior
There should no white bar at the bottom of the page.

### Actual Behavior
There is a white bar at the bottom of the page.

### Basic Information

- Version with issue: 4.4.0.991757
- IDE: 16.4.5
- Platform Target Frameworks: 
  - iOS:  11
- Affected Devices: iPhone 8

**Comment by Mikilll94** [comment]
Here is the repro project

[SafeAreaBug.zip](https://github.com/xamarin/Xamarin.Forms/files/4299772/SafeAreaBug.zip)

**Comment by Mikilll94** [comment]
@hartez 
Is there any workaround for this? I need this urgently.

**Comment by Mikilll94** [comment]
@samhouts @jfversluis 
Please! Do you have any workaround for this bug?

**Comment by jfversluis** [comment]
Of course I don't know the details of your actual app, but if the bug only appears when the NavigationBar is hidden, I guess we would want to work around that. Would it be possible to maybe push a modal with this page that doesn't have a NavigationBar?

Or write a custom renderer that fixes this, or all the way out of the box; use Essentials to detect if it's a iPhone X variant and set the UseSafeLayout to true or false depending on that.
That's some of the ideas that are top of mind. No idea if any of them are feasible

**Comment by Mikilll94** [comment]
@jfversluis 
Do you have any idea where is the source of this bug? I have checked "UpdateUseSafeArea" function in `PageRenderer.cs' but it seems that it is working correctly. For landscape orientation it returns Padding 0,0,0,0

I have no idea how this white bar appears in landscape orientation.

**Comment by jfversluis** [comment]
Unfortunately not. Looking at the screenshots again though, there is something else I'm noticing. The bottom whitespace seems the same height as the statusbar, right? And the statusbar is not visible in landscape. So maybe we were looking into the wrong thing?

**Comment by Mikilll94** [comment]
@jfversluis 

> Unfortunately not. Looking at the screenshots again though, there is something else I'm noticing. The bottom whitespace seems the same height as the statusbar, right? And the statusbar is not visible in landscape. So maybe we were looking into the wrong thing?

Yes, exactly. The bottom whitespace has the same height as the status bar. And the status is not visible in landscape.

BUT on iPhone 8 status bar is the part of Safe Area. Look at this:

![image](https://user-images.githubusercontent.com/17653648/78459780-c425e080-76bb-11ea-99f8-0739251d8f73.png)

And here is landscape

![image](https://user-images.githubusercontent.com/17653648/78459793-dd2e9180-76bb-11ea-9720-2605c5427e3e.png)

And this bug only exists on iOS 13 when I turn on Safe Area for the page. This bug does not exist on iOS <= 12. This is because from iOS 13 you cannot manually adjust the status bar. You can do this only via Safe Area.

So IMO this most probably the bug with Safe Area. Thus `PageRenderer.cs` seems for me a good place when this bug can be fixed. What do you think?

**Comment by elvisxia** [comment]
@jfversluis  Similar issues happens on UWP

**Steps to Reproduce**
1. Create a new project using Xamarin.Forms Tabbed Page template 
2. Add `NavigationPage.HasNavigationBar="False"` to all pages.
3. Run the project and switch to About page and switch back to Items page.

**Expected Behavior**
Navigationbar should be hidden and there is no white space like this:
![image](https://user-images.githubusercontent.com/7528847/80931194-bb274c80-8dea-11ea-9769-f271c963f21f.png)

**Actual Behavior**
NavigationBar is hidden but there is a white space there after switching back to items page:

![image](https://user-images.githubusercontent.com/7528847/80931210-d2fed080-8dea-11ea-8a1f-1243e287c9f4.png)
 

**Basic Information**
Version with issue: 4.4.0.991265
IDE: Visual Studio 2019 on Windows 16.5.4
Platform Target Frameworks:
UWP Build 16299

**Comment by AlexanderMelchers** [comment]
Is there any information on when this will be fixed?

I believe I just ran into a variant of this issue. As this is in a commercial and proprietary app, I, unfortunately, can't post any screen shots, nor is it easy to copy the custom-made control we made and with which we're experiencing the issue into a separate app. As such, I'll do my best to describe the issue:

We've built a custom control that presents the user with a simple pop-over menu, similar to the pop-up below:
![image](https://user-images.githubusercontent.com/28742164/109677620-22806b00-7b7a-11eb-8766-4b3e3e7eac05.png)

This control makes use of a renderer that uses the control's `NextResponder` to `PresentViewController`. This `UIViewController` is a XF `ContentPage` that has the application's main page as its parent and is rendered by Xamarin Forms. All works well on iOS versions prior to 14 (13 not tested, but likely to also have issues), and on devices with a notch (i.e. > iPhone 8).

On iPhone 8, however, the page is offset too much to the top, at the start of the drop-down arrow, with padding also having been added at the bottom of the pop-over. Hard-coding 20 padding into the page pushes the page content down too much, while still maintaining the bottom padding. Turning safe-area insets on for the page, however, correctly aligns the top of the pop-over content, but retains a bottom padding. Turning the safe-area insets on doesn't seem to affect the control on any of the other devices/OS-versions.

I therefore suspect the issue lies with the safe-area insets on iPhone 8 specifically (the only model to _run iOS > 12 to not have a notch_), which seems to match this issue. My version of Xamarin Forms is 5.0.0.1931.

**Comment by AlexanderMelchers** [comment]
> Do you have any idea where is the source of this bug? I have checked "UpdateUseSafeArea" function in `PageRenderer.cs' but it seems that it is working correctly. For landscape orientation it returns Padding 0,0,0,0

Similarly, I'm also getting zero `Thickness` when I query `myPage.On<Xamarin.Forms.PlatformConfiguration.iOS>().SafeAreaInsets()`.

**Comment by AlexanderMelchers** [comment]
My bug seems to go back at least until Forms version 2.5.0.280555...
